src/odemis/acq/acqmng.py

```diff
@@ -83,6 +85,202 @@ def acquire(streams, settings_obs=None):
     return future
 
 
+def acquireZStack(streams, zlevels, settings_obs=None):
+    """
+    The acquisition manager of a zstack
+    streams (list of Stream): the streams to be acquired
+    zlevels (dict Stream -> list of floats): a dictionary containing the streams and the
+    corresponding lists of actuator z positions
+    settings_obs (SettingsObserver): VAs to be integrated in the acquired data as metadata
+    return (ProgressiveFuture): the future that will be executing the task
+    """
+    # create future
+    future = model.ProgressiveFuture()
+    # create acquisition task
+    acqui_task = ZStackAcquisitionTask(future, streams, zlevels, settings_obs)
+    # add the ability of cancelling the future during execution
+    future.task_canceller = acqui_task.cancel
+
+    # set the progress of the future
+    total_duration = acqui_task.estimate_total_duration()
+    future.set_end_time(time.time() + total_duration)
+
+    # assign the acquisition task to the future
+    executeAsyncTask(future, acqui_task.run)
+
+    return future
+
+
+class ZStackAcquisitionTask(object):
+    """
+    This class represents an acquisition task for a zstack of data.
+    """
+
+    def __init__(self, future, streams, zlevels, settings_obs):
+        """
+        The class constructor
+        future (ProgressiveFuture): the future that will execute the task asynchronously
+        streams (list of Stream): the streams to be acquired
+        zlevels (dict Stream -> list of floats): a dictionary containing the streams and the
+        corresponding lists of actuator z positions
+        settings_obs (SettingsObserver): VAs to be integrated in the acquired data as metadata
+        """
+        self._main_future = future
+        self._future_lock = threading.Lock()
+        self._actuator_f = None 
+        self._single_acqui_f = None 
+        self._future_state = RUNNING 
+        self._streams = sortStreams(streams)
+        self._zlevels = zlevels
+        self._settings_obs = settings_obs
+        self._zstep_duration = {}
+        for s,z in zlevels.items():
+            self._zstep_duration[s] = guessActuatorMoveDuration(s.focuser, axis="z", distance=abs(z[0]-z[1]))
```

Comment:
Please, mind the spaces. 
For commas: 1 space after, such as `a, b` .
For operators: space before & after, such as `a - b`.